polygeist/tools/cgeist/Test/Verification/compoundassign.c

```diff
@@ -171,3 +173,64 @@ unsigned f11(unsigned *a, unsigned b) {
 unsigned f12(unsigned *a, unsigned b) {
   return *a ^= b;
 }
+
+// COM: This test is incorrect due to signedness detection issues.
+
+// CHECK-LABEL:   func.func @f13(
+// CHECK-SAME:                   %[[VAL_0:.*]]: memref<?xi8>,
+// CHECK-SAME:                   %[[VAL_1:.*]]: i32) -> i8
+// CHECK:           %[[VAL_2:.*]] = arith.constant 0 : i32
+// CHECK:           %[[VAL_3:.*]] = affine.load %[[VAL_0]][0] : memref<?xi8>
+// CHECK:           %[[VAL_4:.*]] = arith.extui %[[VAL_3]] : i8 to i32
+// CHECK:           %[[VAL_5:.*]] = arith.addi %[[VAL_4]], %[[VAL_1]] : i32
+// CHECK:           %[[VAL_6:.*]] = arith.cmpi ne, %[[VAL_5]], %[[VAL_2]] : i32
+// CHECK:           %[[VAL_7:.*]] = arith.extui %[[VAL_6]] : i1 to i8
+// CHECK:           affine.store %[[VAL_7]], %[[VAL_0]][0] : memref<?xi8>
+// CHECK:           return %[[VAL_7]] : i8
+// CHECK:         }
+
+bool f13(bool *a, unsigned b) {
+  return *a += b;
+}
+
+// COM: Currently not working for how +operator is implemented for pointers.
+
+#if 0
+bool f14(bool *a, void *b) {
+  return *a += b;
+}
+#endif
+
+// CHECK-LABEL:   func.func @f15(
+// CHECK-SAME:                   %[[VAL_0:.*]]: memref<?xi8>,
+// CHECK-SAME:                   %[[VAL_1:.*]]: f32) -> i8
+// CHECK:           %[[VAL_2:.*]] = arith.constant 0.000000e+00 : f32
+// CHECK:           %[[VAL_3:.*]] = affine.load %[[VAL_0]][0] : memref<?xi8>
+// CHECK:           %[[VAL_4:.*]] = arith.uitofp %[[VAL_3]] : i8 to f32
+// CHECK:           %[[VAL_5:.*]] = arith.addf %[[VAL_4]], %[[VAL_1]] : f32
+// CHECK:           %[[VAL_6:.*]] = arith.cmpf une, %[[VAL_5]], %[[VAL_2]] : f32
+// CHECK:           %[[VAL_7:.*]] = arith.extui %[[VAL_6]] : i1 to i8
+// CHECK:           affine.store %[[VAL_7]], %[[VAL_0]][0] : memref<?xi8>
+// CHECK:           return %[[VAL_7]] : i8
+// CHECK:         }
+
+bool f15(bool *a, float b) {
+  return *a += b;
+}
+
+// CHECK-LABEL:   func.func @f16(
+// CHECK-SAME:                   %[[VAL_0:.*]]: memref<?xi8>,
+// CHECK-SAME:                   %[[VAL_1:.*]]: f64) -> i8
```

Comment:
i8 -> i1

Reply:
Same as above, returning `i1` here would require too many changes to include in a single PR.

Reply:
Ok I see, wasn't clear to me whether the fix is relatively simple or complex. In that case perhaps is best to open an issue and address the issue in a separate pull request.